docs/discussion/Ratify v2 design scope.md

```diff
@@ -0,0 +1,126 @@
+2. Update the Oras store configuration format so that it can support multiple auth providers in the future.
+
+#### Reorganize the Codebase
+**Current Limitation**:
+1. We have everything in the same repo which adds the dependencies of the project. e.g. we have external plugins' implementation in the Ratify repo. 
+2. The overall codebase is not well organized. We exposed some internal packages to customers and it's difficult to understand the functionality/difference of each package.
+
+**Proposed Solution**:
+1. Move the external plugins to a separate repo. This will reduce the dependencies of the project.
+2. Reorganize the codebase. We should have a clear separation of the internal and external packages. We should also have a clear separation of the core/extended functionality, and CLI/Stand-alone service/K8s service.
+
+### Deprecations
+#### CertificateStore
+**Current Limitation**:
+1. `CertificateStore` is a previous version of `KeyManagementProvider`. It's incompatible with multi-tenancy scenario and does not support the latest features of the `KeyManagementProvider`, such as key rotation.
+
+**Proposed Solution**:
+1. Deprecate `CertificateStore` and recommend customers to use `KeyManagementProvider` instead.
+
+#### Legacy Cosign Verifier
+**Current Limitation**:
+1. The legacy Cosign verifier does not support referencing multiple keys and KeyManagementProvider.
+2. The legacy Cosign verifier does not support trust policy for cosign verification.
+
+**Proposed Solution**:
+1. Deprecate the legacy Cosign verifier and recommend customers to use the new Cosign verifier.
+
+#### License Checker Verifier
+**Current Limitation**:
+1. The primitive implementation of the licensechecker verifier to support basic verification of license compliance (allowed license list).
+
+**Proposed Solution**:
+1. Deprecate the licensechecker verifier and recommend customers to use the new SBOM verifier.
+
+#### `Name` and `Type` fields in verifierReport/configuration
+**Current Limitation**:
+1. `Name` and `Type` fields in VerifierReport refer to the name and type of the verifier that generated the report. But we got feedback that they're ambiguous and can be misleading.
+2. `Name` and `Type` fields in plugin configuration are misused due to historical reasons. It makes the configuration hard to understand.
+
+**Proposed Solution**:
+1. Deprecate `Name` and `Type` fields in VerifierReport and recommend customers to use `VerifierName` and `VerifierType` instead.
+2. Correct the usage of `Name` and `Type` fields in plugin configuration.
+
+### New Features
+
+The v2 design mainly focuses on the fundamental refactoring and deprecations that are necessary for the system. In our first prototype or first RC/official release, we may not have new features. However, we should keep the following features in mind to avoid introducing new limitations or breaking changes in the future.
```

Comment:
Looking at the whole proposal, it seems primarily focus on the refactoring and deprecation but less on new features. The title seems like look ahead a BIG plan of a new major milestone (v2) but without such detailed context as refactoring and deprecation.  

How about focusing on the refactoring and deprecation only, moving the feature planning to the ROADMAP.md? It would be good to add concrete plans for the whole v2 with each minor milestone (v2.x) in a central roadmap doc.

Reply:
Thereby, I am thinking this title "Proposal: A more lightweight and extensible framework for Ratify v2". 
Deprecation makes Ratify more lightweight and refactoring is intended for evolving to be a more maintainable and extensible framework.

Reply:
Good call, my initial idea to add the new feature section is to remind me of those features while refactoring. I would remove it if more reviewers agree on it. And I can also just add link to our roadmap if necessary.